repose-aggregator/services/atom-feed/impl/src/main/scala/org/openrepose/nodeservice/atomfeed/impl/AtomFeedServiceImpl.scala

```diff
@@ -0,0 +1,228 @@
+              actorSystem.actorOf(FeedReader.props(
+                feedConfig.getUri,
+                (arf: ActorRefFactory) =>
+                  arf.actorOf(Authenticator.props(feedConfig.getAuthentication)),
+                (arf: ActorRefFactory) =>
+                  arf.actorOf(Notifier.props(feedListeners.getOrElse(feedId, Map.empty).values.toSet)),
+                feedConfig.getEntryOrder
+              )),
+              ReadFeed
+            )
+            feedSchedules = feedSchedules + (feedId -> schedule)
+          }
+
+          listenerId
+        case None =>
+          throw new IllegalArgumentException("feedId parameter does not match a configured feed id")
+      }
+    }
+  }
+
+  override def unregisterListener(listenerId: String): Unit = {
+    ifRunning {
+      feedListeners find {
+        case (_, listeners) => listeners.contains(listenerId)
+      } match {
+        case Some((feedId, listeners)) =>
+          logger.debug("Un-registering the listener: " + listenerId)
+
+          val newListeners = listeners filterNot { case (id, listener) => listenerId.equals(id) }
+          feedListeners = feedListeners + (feedId -> newListeners)
+          if (newListeners.isEmpty) {
+            feedSchedules(feedId).cancel()
+          }
+        case None =>
+          logger.debug("Listener not registered: " + listenerId)
+      }
+    }
+  }
+
+  override def isRunning: Boolean = running
+
+  private def startService(): Unit = {
+    synchronized {
+      if (isInitialized && isServiceEnabled && !running) {
+        logger.info("Starting the service")
+        actorSystem = ActorSystem.create("AtomFeedServiceSystem")
+        actorSystem.registerOnTermination(running = false)
+        running = true
+      } else if (running) {
+        logger.debug("Service not starting -- service is already running")
+      } else if (!isServiceEnabled) {
+        logger.debug("Service not starting -- disabled by configuration")
+      } else {
+        logger.debug("Service not starting -- waiting on configuration files to load")
+      }
+    }
+  }
+
+  private def stopService(): Unit = {
+    synchronized {
+      if (running) {
+        logger.info("Stopping the service")
+        actorSystem.shutdown()
+        actorSystem.awaitTermination(5 seconds)
+      } else {
+        logger.debug("Service not stopping -- service is already stopped")
+      }
+    }
+  }
+
+  private def ifRunning[T](f: => T): T = {
+    if (running) f
+    else throw new IllegalStateException("Service is not running")
+  }
+
+  private def getListenersForFeed(feedId: String): Map[String, AtomFeedListener] = {
+    feedListeners.getOrElse(feedId, Map.empty)
+  }
+
+  def isInitialized: Boolean = {
+    SystemModelConfigurationListener.isInitialized && AtomFeedServiceConfigurationListener.isInitialized
+  }
+
+  object AtomFeedServiceConfigurationListener extends UpdateListener[AtomFeedServiceConfigType] {
```

Comment:
this is broken. you've restarted the actor system, but all the current listeners are not hooked into